datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -43,24 +43,35 @@ type ChainDataFetcher struct {
 
 	numHandlers int
 
+	checkpoint    int64
+	checkpointMap map[int64]struct{}
+
 	wg sync.WaitGroup
 
-	repo repository
+	repo Repository
 }
 
 func NewChainDataFetcher(ctx *node.ServiceContext, cfg *ChainDataFetcherConfig) (*ChainDataFetcher, error) {
 	repo, err := kas.NewRepository(cfg.DBUser, cfg.DBPassword, cfg.DBHost, cfg.DBPort, cfg.DBName)
 	if err != nil {
+		logger.Error("Failed to create new Repository", "err", err, "user", cfg.DBUser, "password", cfg.DBPassword, "host", cfg.DBHost, "port", cfg.DBPort, "name", cfg.DBName)
```

Comment:
Please delete password from the log.

Reply:
Removed password log. 3e445f1a1a797cfaa848ef462b0fc59b87bec994